### ISSUE TITLE
Path conversion doesn't always work on paste

### ISSUE THREAD

**Issue by kenijo** [mintty/wsltty#161]
I have been able to drag and drop a file from explorer to wsltty and the path conversion works great.
I haven't been able to paste a path (shift+insert or mouse right click) and get the conversion to work.

Is it possible to add this function? wslbridge included in ConEmu handles it fine so I am guessing it is not implemented in the wsltty version.

**Comment by Biswa96** [comment]
Which path conversion are you talking about?  Provide an example.

1. Win32 <--> Cygwin
2. Win32 <--> WSL

**Comment by kenijo** [comment]
I would say Win32 <-> WSL since I posted the question in wsltty. 
But if you are asking the question  maybe I am missing something else.

**Comment by mintty** [comment]
Path conversion is applied only for drag-and-drop (or, reversely, for Control+click link activation).
When pasting text, it is not applied; you could simply wish to quote something copied from a plain text file, which happens to be a path name; I see no reasonable way to decide whether it's meant as a working path.

**Comment by mintty** [comment]
See also https://github.com/mintty/mintty/issues/785#issuecomment-410449222.

**Comment by kenijo** [comment]
Thank you for pointing me to this comment, very informative. I couldn't fin it yesterday.

I think if you consider that WSL as its own environment and everything that happens in there is supposed to be WSL related only (hence pasting a path inside wsltty should be wsl formated) then it could make a case for it. I see value in both approach.

Maybe an option to enable path conversion on pasting could be viable?

Would running everything pasted through wslpath to convert it to WSL format (and everything copied to convert it to Win32 format) impact the application performances to much?

**Comment by mintty** [comment]
Actually my previous comment was wrong. Whether path conversion is applied depends on the property tag of the Windows clipboard. So if you copy e.g. a folder, its name will be path-converted, whether you use drag-and-drop, Shift+Ins or Paste from the menu. I'll check whether that also works in WSL.
That means if you select some text that happens to be a path name, however you paste it, it will not be converted. I think this makes my previous argument and the comment I referred to even more valid.
If you have another idea, please describe your preferred design precisely.

**Comment by kenijo** [comment]
Here is what I tried in wsltty:
-  copy a file itself in Windows Explorer (CTRL+C or right click) and paste in WSLTTY: `C:\myfile.txt` turns into `/mnt/c/myfile.txt`
- copy a path from the Windows Explorer location bar and paste in WSLTTY: `C:\myfile.txt` stays as is.

I tried the same in ConEmu and the results are almost opposite:
-  copy a file itself in Windows Explorer (CTRL+C or right click) and paste in WSLTTY: `C:\myfile.txt` doesn't do anything.
- copy a path from the Windows Explorer location bar and paste in WSLTTY: `C:\myfile.txt` turns into `/mnt/c/myfile.txt`.

Both programs use the wslbridge-backend but with different results.

Anyway, my idea (and I have no idea if this is doable) was to send intercept copy / paste events and send it through wslpath such as:
- paste in WSLTTY -> wslpath -u myargument
If the result triggers an error then paste as the original (this means "myargument" was not a Windows path); otherwise the path got properly converted.
- copy from WSLTTY -> wslpath -w myargument
If the result triggers an error then copy as the original (this means "myargument" was not a Unix path); otherwise the path got properly converted.

Again, I am just throwing an idea here as I don't have a very good understanding of how all this works.
